server/server.go

```diff
@@ -27,24 +26,24 @@ import (
 
 // Publisher contains all operations required for the shipper server to publish incoming events.
 type Publisher interface {
-	io.Closer
-
-	// AcceptedIndex returns the current sequential index of the accepted events
-	AcceptedIndex() queue.EntryID
 	// PersistedIndex returns the current sequential index of the persisted events
-	PersistedIndex() queue.EntryID
-	// Publish publishes the given event and returns the current accepted index (after this event).
-	// This operation is blocking until the event is published or the target queue is closed.
-	Publish(context.Context, *messages.Event) (queue.EntryID, error)
-	// TryPublish tries to immediately publish the given event and returns the current accepted index (after this event).
-	// Returns an error if it was not possible to publish the event without blocking.
-	TryPublish(*messages.Event) (queue.EntryID, error)
+	PersistedIndex() (queue.EntryID, error)
+
+	// Publish publishes the given event and returns its index.
+	// It blocks until the event is published or the given context is canceled
+	// (or the target queue is closed).
+	Publish(ctx context.Context, event *messages.Event) (queue.EntryID, error)
+
+	// TryPublish publishes the given event and returns its index.
+	// If the event cannot be published without blocking, TryPublish returns an error.
+	TryPublish(event *messages.Event) (queue.EntryID, error)
 }
 
 // ShipperServer contains all the gRPC operations for the shipper endpoints.
 type ShipperServer interface {
+	Close() error
```

Comment:
I think it would be more conventional to embed `io.Closer` here instead. But it's more like nitpicking.

Reply:
That is a common convention but I prefer to include the `Close` explicitly: if you use `io.Closer`, and then use the IDE's "find all references" on one of the calls, it will find all `Close` calls to _any_ `io.Closer`, whereas if we make the function explicit, it will still satisfy the `io.Closer` interface but searching for all references will produce only the `Close` calls specifically on a `ShipperServer`.